Automattic/jetpack
# Consolidate the Lazy Images package to rely on the Assets package

Fixes #16971 

#### Changes proposed in this Pull Request:
* Make the Lazy Images package dependent on the Assets package in order to make use of its `get_file_url_for_environment` method to the the asset file url for enqueuing
* Refactor the `get_file_url_for_environment` method from the Assets package to add a plugin path optional parameter to allow get the file url  relative to a package directory rather than the root Jetpack directory.

#### Does this pull request change what data or activity we track or use?
No

#### Testing instructions:
- Go to wp-admin/admin.php?page=jetpack_modules
- Activate the "Lazy images" module
- Create a new post with an image inside it.
- Open that post as a non logged in user
- Look at the source of the page and verify that the img tag as the data-lazy-loaded="1" property.

#### Proposed changelog entry for your changes:

* Consolidate the Lazy Images package to rely on the Assets package to get the asset file url.
